python/pyspark/sql/functions.py

```diff
@@ -33,17 +33,36 @@
 __all__ = ['countDistinct', 'approxCountDistinct', 'udf']
 
 
-def _create_function(name, doc=""):
+def _function_obj(sc, is_math=False):
+    if not is_math:
+        return sc._jvm.functions
+    else:
+        return sc._jvm.mathfunctions
+
+
+def _create_function(name, doc="", is_math=False, binary=False):
     """ Create a function for aggregator by name"""
-    def _(col):
+    def _(col1, col2=None):
```

Comment:
It is somewhat strange to have col2 be default None. I think it's easier if we just create a _create_binary_function function.